File: ios/Runner/Repository/Contract/AlbumRepositoryContract.swift
//
//  AlbumRepositoryContract.swift
//  Runner
//
//  Created by t k on 2024/08/12.
//

protocol AlbumRepositoryContract {
    func findAll() -> [AlbumModelContract]

    func findByAlbumId(albumId: AlbumId) -> AlbumModelContract?

    func findByArtistId(artistId: ArtistId) -> [AlbumModelContract]

    func findFirstByArtistId(artistId: ArtistId) -> AlbumModelContract?

    func findRecentAlbums(count: Int) -> [AlbumModelContract]
}

File: android/app/src/main/kotlin/dev/tcode/thinmpf/repository/contract/AlbumRepositoryContract.kt
package dev.tcode.thinmpf.repository.contract

import dev.tcode.thinmpf.model.contract.AlbumModelContract
import dev.tcode.thinmpf.model.valueObject.AlbumId
import dev.tcode.thinmpf.model.valueObject.ArtistId

interface AlbumRepositoryContract {
    fun findAll(): List<AlbumModelContract>

    fun findByAlbumId(albumId: AlbumId): AlbumModelContract?

    fun findByArtistId(artistId: ArtistId): List<AlbumModelContract>

    fun findFirstByArtistId(artistId: ArtistId): AlbumModelContract?

    fun findRecentAlbums(limit: Int): List<AlbumModelContract>
}